src/robot/libraries/BuiltIn.py

```diff
@@ -745,87 +761,154 @@ def should_be_equal_as_numbers(self, first, second, msg=None, values=True,
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``str1`` and ``str2`` should be compared case-insensitively,
+        provided that ``str1`` and ``str2`` are string types.  See `Boolean
+        arguments` section for more details.  (New in RF 3.0.1)
+
         See `Should Be Equal` for an explanation on how to override the default
         error message with ``msg`` and ``values``.
         """
+        if is_truthy(ignore_case) and is_string(str1) and is_string(str2):
+            str1 = str1.lower()
+            str2 = str2.lower()
         if str1.endswith(str2):
             raise AssertionError(self._get_string_msg(str1, str2, msg, values,
                                                       'ends with'))
 
-    def should_end_with(self, str1, str2, msg=None, values=True):
+    def should_end_with(self, str1, str2, msg=None, values=True, ignore_case=False):
         """Fails if the string ``str1`` does not end with the string ``str2``.
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``str1`` and ``str2`` should be compared case-insensitively,
+        provided that ``str1`` and ``str2`` are string types.  See `Boolean
+        arguments` section for more details.  (New in RF 3.0.1)
+
         See `Should Be Equal` for an explanation on how to override the default
         error message with ``msg`` and ``values``.
         """
+        if is_truthy(ignore_case) and is_string(str1) and is_string(str2):
+            str1 = str1.lower()
+            str2 = str2.lower()
         if not str1.endswith(str2):
             raise AssertionError(self._get_string_msg(str1, str2, msg, values,
                                                       'does not end with'))
 
-    def should_not_contain(self, container, item, msg=None, values=True):
+    def should_not_contain(self, container, item, msg=None, values=True, ignore_case=False):
         """Fails if ``container`` contains ``item`` one or more times.
 
         Works with strings, lists, and anything that supports Python's ``in``
         operator. See `Should Be Equal` for an explanation on how to override
         the default error message with ``msg`` and ``values``.
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``container`` and ``item`` should be compared case-insensitively,
+        provided that ``container`` and ``item`` are string types.  See `Boolean
+        arguments` section for more details.  (New in RF 3.0.1)
+
         Examples:
         | Should Not Contain | ${output}    | FAILED |
         | Should Not Contain | ${some list} | value  |
         """
-        if item in container:
+        prv_container = container
+        prv_item = item
+        if is_truthy(ignore_case) and is_string(container) and is_string(item):
+            prv_container = container.lower()
+            prv_item = item.lower()
+        if prv_item in prv_container:
```

Comment:
As I already wrote in the previous comment, we should also support a case that the container is a list or list-like. In my opinion we should convert all strings in the container to lower case first. Something like this:

```python
tested_container = container
if is_truthy(ignore_case) and is_string(item):
    item = item.lower()
    if is_list_like(container):    # needs to be imported from `robot.utils`
        tested_container = set([c.lower() if is_string(c) else c for c in container])
if item in test_container:
    # ...
```

The above implementation converts all list-like containers (including dictionaries) to sets (could also use lists, and could use set literals if we still didn't need to support Python 2.6). Need to preserve the original container too to be able to show the original structure in possible error message.

In theory `item` could also be a list of strings and `container` could be a nested list of strings, but recursively lowercasing everything is probably not a good idea. It would add quite a bit of complexity, including making sure we don't change the type of the `item`.